core/regions/src/main/java/software/amazon/awssdk/regions/servicemetadata/EnhancedS3ServiceMetadata.java

```diff
@@ -100,12 +100,16 @@ private static String envVarSetting() {
         return SdkSystemSetting.AWS_S3_US_EAST_1_REGIONAL_ENDPOINT.getStringValue().orElse(null);
     }
 
-    private String profileFileSetting(Supplier<ProfileFile> profileFileSupplier, Supplier<String> profileName) {
+    private String profileFileSetting(Supplier<ProfileFile> profileFileSupplier, Supplier<String> profileNameSupplier) {
         try {
-            return profileFileSupplier.get()
-                                      .profile(profileName.get())
-                                      .flatMap(p -> p.property(ProfileProperty.S3_US_EAST_1_REGIONAL_ENDPOINT))
-                                      .orElse(null);
+            ProfileFile profileFile = profileFileSupplier.get();
+            String profileName = profileNameSupplier.get();
+            if (profileFile == null || profileName == null) {
+                return null;
+            }
+            return profileFile.profile(profileName)
+                              .flatMap(p -> p.property(ProfileProperty.S3_US_EAST_1_REGIONAL_ENDPOINT))
+                              .orElse(null);
```

Comment:
This change is unrelated, but it prevents an error message being logged when the profile file is null.